server/routerlicious/packages/lambdas/src/scriptorium/lambda.ts

```diff
@@ -87,16 +99,40 @@ export class ScriptoriumLambda implements IPartitionLambda {
         Promise.all(allProcessed).then(
             () => {
                 this.current.clear();
-                // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-                this.context.checkpoint(batchOffset!);
+                status = "ProcessingComplete";
+
+                // checkpoint batch offset
+                try {
+                    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+                    this.context.checkpoint(batchOffset!);
+                    status = "CheckpointComplete";
+                } catch (error) {
```

Comment:
This catch intercepted all exceptions and changed the flow, is this desired?

Reply:
Indeed, we might want to rethrow the error after logging on line 113

Reply:
Thanks @zhangxin511 and @hedasilv . I removed this catch block and will let KafkaRunner metric log this error message.

Reply:
@shubhi1092 - but without the catch block, we run the risk of not completing the metric in case of exceptions (problem mentioned here: https://github.com/microsoft/FluidFramework/pull/14179#discussion_r1106473788), no?

Reply:
My suggestion above - which addresses the issue Xin mentioned - is to still keep the catch block, allowing you to complete the metric, and then you rethrow the error to make sure the newly-introduced catch block does not swallow the exception

Reply:
Yes @hedasilv , in that case the scriptorium metric wont complete -- for my understanding, does that cause issues other than the fact that it wont be logged in kusto?
Also looking at the code for `context.checkpoint`, it seems that it is catching the exception itself and will not throw it outside -- https://github.com/microsoft/FluidFramework/blob/main/server/routerlicious/packages/lambdas-driver/src/kafka-service/context.ts#L25
Let me know if I am misunderstanding.

Reply:
The metric might not be that helpful if it does not log certain failures... It is true that the checkpoint method might eventually capture the failures here, but if that's already the case, then one could ask: why do we need this scriptorium handler metric at all?
In my opinion, there is value in adding that metric, but we need to make sure it is logged in all cases. My suggestion would be to add the try/catch again, and in the `catch (error) { }` block, just add a `throw error;` at the end of the block. I think that is the best approach here

Reply:
Catch, terminate the metric in error state, and rethrow would keep the flow the same as before, and make sure the metric terminates correctly.

Also, no need to do an error logging + metric.error together. `metric.error` would do the error logging for your automatically